src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8140,130 +8157,197 @@ void checkWellKnown(WellKnownMember member)
+                                candidate is PropertySymbol property &&
+                                IsAccessible(property, syntax, diagnostics) &&
+                                property.OriginalDefinition is { ParameterCount: 1 } original &&
+                                original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                            {
+                                var intPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
+                                argumentPlaceholders = ImmutableArray.Create(intPlaceholder);
+
+                                var analyzedArguments = AnalyzedArguments.GetInstance();
+                                analyzedArguments.Arguments.Add(intPlaceholder);
+                                var properties = ArrayBuilder<PropertySymbol>.GetInstance();
+                                properties.AddRange(property);
+                                indexerOrSliceAccess = BindIndexerOrIndexedPropertyAccess(syntax, receiver, properties, analyzedArguments, diagnostics).MakeCompilerGenerated();
+                                properties.Free();
+                                analyzedArguments.Free();
+                                lookupResult.Free();
+                                return true;
+                            }
                         }
                     }
                 }
-            }
-            else if (receiverType.SpecialType == SpecialType.System_String)
-            {
-                Debug.Assert(!argIsIndex);
-                // Look for Substring
-                var substring = (MethodSymbol)Compilation.GetSpecialTypeMember(SpecialMember.System_String__Substring);
-                // PROTOTYPE(list-patterns) Consider reporting a missing member for string.Substring
-                if (substring is object)
+                else if (receiver.Type.SpecialType == SpecialType.System_String)
                 {
-                    patternSymbol = substring;
-                    return true;
+                    Debug.Assert(!argIsIndex);
+                    // Look for Substring
+                    var substring = (MethodSymbol)Compilation.GetSpecialTypeMember(SpecialMember.System_String__Substring);
+                    if (substring is object)
+                    {
+                        makeCall(syntax, receiver, substring, out indexerOrSliceAccess, out argumentPlaceholders);
+                        lookupResult.Free();
+                        return true;
+                    }
                 }
-            }
-            else
-            {
-                Debug.Assert(!argIsIndex);
-                // Look for `T Slice(int, int)` indexer
+                else
+                {
+                    Debug.Assert(!argIsIndex);
+                    // Look for `T Slice(int, int)` indexer
```

Comment:
It feels like we should just be using the `MakeInvocationExpression` helper here, rather than replicating lookup and overload resolution logic.

Reply:
> It feels like we should just be using the MakeInvocationExpression helper here, rather than replicating lookup and overload resolution logic.

It looks like ```MakeInvocationExpression``` doesn't provide the same level of control over what is considered an applicable candidate as we have here. Also, for string type we don't even use regular lookup.